mParticle-Example/MPKitExample.m

```diff
@@ -30,10 +30,20 @@ - (MPKitExecStatus *)execStatus:(MPKitReturnCode)returnCode {
 
 #pragma mark Kit instance and lifecycle
 - (MPKitExecStatus *)didFinishLaunchingWithConfiguration:(NSDictionary *)configuration {
-    NSString *apiKey = configuration[@"<dictionary key to retrieve API Key>"];
-    if (!apiKey) {
+    NSString *consumerKey = configuration[PILGRIM_SDK_KEY];
```

Comment:
these also could have stronger type safety if you make helper functions to guarantee type:
`stringOrNil(configuration[PILGRIM_SDK_KEY])`
`numberOrNil(configuration[PERSIST_LOGS])`

where the implementation would look something like:
```
NS_INLINE NSString* _Nullable stringOrNil(id _Nullable value) {
    return [value isKindOfClass:[NSString class]] ? (NSString *)value : nil;
}
```

Reply:
Great idea! I put it in. However for the persist logs, the value isn't a nsnumber, so is there a way to confirm it's a bool since it's a primitive?